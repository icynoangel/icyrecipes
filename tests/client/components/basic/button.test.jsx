import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import Button from './../../../../src/client/js/components/basic/button';
import renderer from 'react-test-renderer';

describe('<Button />', function() {
  this.getComponent = type => {
    return <Button onClick={this.onClick} caption={this.caption} type={type} />;
  };

  beforeEach(() => {
    this.onClick = jest.fn();
    this.caption = 'Get Items';
  });

  it('Should render a primary Button', () => {
    const button = renderer.create(this.getComponent('primary')).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('Should render a secondary Button', () => {
    const button = renderer.create(this.getComponent('secondary')).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('Should call onClick function', () => {
    const wrapper = shallow(this.getComponent('secondary'));
    wrapper.find('button').simulate('click');
    expect(this.onClick).toHaveBeenCalledTimes(1);
  });
});
