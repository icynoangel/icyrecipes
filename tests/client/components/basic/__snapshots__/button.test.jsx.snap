// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render a primary Button 1`] = `
<button
  className="button --primary"
  onClick={[Function]}
>
  <span
    className="button__caption"
  >
    Get Items
  </span>
</button>
`;

exports[`<Button /> Should render a secondary Button 1`] = `
<button
  className="button --secondary"
  onClick={[Function]}
>
  <span
    className="button__caption"
  >
    Get Items
  </span>
</button>
`;
